### PR TITLE
Create CircuitBreaker class and tests

### PR DESCRIPTION
**Pull request by vtrmantovani**
Adding CircuitBreaker class in project

In this pr i am creating only the base class of circuit breaker, after then I will open two pr separate to support async with await and yield.

 ### Checklist:
 - [x] Tests
 - [ ] Changelog
 - [ ] Docs

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+	@awk 'BEGIN {FS = ":.*?## "} /^[a-zA-Z_-]+:.*?## / {printf "\033[36m%-20s\033[0m %s\n", $$1, $$2}' $(MAKEFILE_LIST)
+
+clean: ## Clean cache and temporary files
+	@find . -name "*.pyc" | xargs rm -rf
+	@find . -name "*.pyo" | xargs rm -rf
+	@find . -name "__pycache__" -type d | xargs rm -rf
+	@rm -rf *.egg-info
+	@rm -f .coverage
+
+check:  ## Run static code checks
```
Comment (luizgois-luizalabs) [inline Makefile]: acho que seria melhor `lint`.

File: asyncio_toolkit/coroutine/fallbacks/circuit_breaker.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
+# -*- coding: utf-8 -*-
```
Comment (luizgois-luizalabs) [inline asyncio_toolkit/coroutine/fallbacks/circuit_breaker.py]: No `python3` não precisa colocar.

File: asyncio_toolkit/coroutine/fallbacks/circuit_breaker.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        if exc_type in self.catch_exceptions:
+            if self.is_circuit_open:
+                raise self.max_failure_exception
+
+            self._increase_failure_count()
+
+            if self.total_failures >= self.max_failures:
+                self.open_circuit()
+
+                logger.info(
```
Comment (luizgois-luizalabs) [inline asyncio_toolkit/coroutine/fallbacks/circuit_breaker.py]: Aqui poderia ser assim:

```
logger.info(
    'Max failures exceeded by: {}'.format(self.failure_cache_key)
)
```

File: Makefile (hunk 1 above)
Comment (vtrmantovani) [reply Makefile]: 👍

File: asyncio_toolkit/coroutine/fallbacks/circuit_breaker.py (hunk 2 above)
Comment (vtrmantovani) [reply asyncio_toolkit/coroutine/fallbacks/circuit_breaker.py]: 👍

File: asyncio_toolkit/coroutine/fallbacks/circuit_breaker.py (hunk 3 above)
Comment (vtrmantovani) [reply asyncio_toolkit/coroutine/fallbacks/circuit_breaker.py]: 👍

**Review by luizgois-luizalabs — APPROVED** [review]

**Comment by Bgouveia** [comment]
@vtrmantovani IMHO a (possibly) better design for the packaging could be `asyncio_toolkit/fallbacks/coroutine` instead of `asyncio_toolkit/coroutine/fallbacks`; in this case, this implementation could be placed in `asyncio_toolkit/fallbacks/circuitbreaker.py`. 
If this is the choice you go for, the implementation of `await` and `yield` would be placed inside their respective modules and the code that implements this piece of code would have to explicitly mention which aspect of asynchronous mechanism it uses.. what do you think?

**Review by Bgouveia — APPROVED** [review]

**Pull request merged**